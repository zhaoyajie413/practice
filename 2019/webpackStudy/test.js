const runCommand = (command, args) => {
    const cp = require("child_process");
    return new Promise((resolve, reject) => {
        const executedCommand = cp.spawn(command, args, {
            stdio: "inherit",
            shell: true
        });

        executedCommand.on("error", error => {
            reject(error);
        });

        executedCommand.on("exit", code => {
            if (code === 0) {
                resolve();
            } else {
                reject();
            }
        });
    });
};
//runCommand('npm',['install','element-ui','-s'])
runCommand('notepad',[])
