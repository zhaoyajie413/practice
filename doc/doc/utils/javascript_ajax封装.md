AJAX 的封装
================

>code:

```$xslt
/**
 * 发起网络请求
 * @param {object} param0
 * @param {string} callbackId
 * @return {RequestTask}
 */
export function request ({
                             url,
                             data,
                             header,
                             method,
                             dataType,
                             responseType
                         }, callbackId) {
    const {
        invokeCallbackHandler: invoke
    } = UniServiceJSBridge
    var body = null
    var timeout = (__uniConfig.networkTimeout && __uniConfig.networkTimeout.request) || 60 * 1000
    // 根据请求类型处理数据
    var contentType
    for (const key in header) {
        if (header.hasOwnProperty(key)) {
            if (key.toLowerCase() === 'content-type') {
                contentType = header[key]
                if (contentType.indexOf('application/json') === 0) {
                    contentType = 'json'
                } else if (contentType.indexOf('application/x-www-form-urlencoded') === 0) {
                    contentType = 'urlencoded'
                } else {
                    contentType = 'string'
                }
                break
            }
        }
    }
    if (method === 'GET') {
        url = setUrl(url, data)
    } else {
        if (!contentType) {
            /**
             * 跨域时部分服务器OPTION响应头Access-Control-Allow-Headers未包含Content-Type会请求失败
             */
            header['Content-Type'] = 'application/json'
            contentType = 'json'
        }
        if (typeof data === 'string' || data instanceof ArrayBuffer) {
            body = data
        } else {
            if (contentType === 'json') {
                try {
                    body = JSON.stringify(data)
                } catch (error) {
                    body = data.toString()
                }
            } else if (contentType === 'urlencoded') {
                let bodyArray = []
                for (let key in data) {
                    if (data.hasOwnProperty(key)) {
                        bodyArray.push(encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
                    }
                }
                body = bodyArray.join('&')
            } else {
                body = data.toString()
            }
        }
    }
    var xhr = new XMLHttpRequest()
    var requestTask = new RequestTask(xhr)
    xhr.open(method, url)
    for (var key in header) {
        if (header.hasOwnProperty(key)) {
            xhr.setRequestHeader(key, header[key])
        }
    }

    var timer = setTimeout(function () {
        xhr.onload = xhr.onabort = xhr.onerror = null
        requestTask.abort()
        invoke(callbackId, {
            errMsg: 'request:fail timeout'
        })
    }, timeout)
    xhr.responseType = responseType.toLowerCase()
    xhr.onload = function () {
        clearTimeout(timer)
        let statusCode = xhr.status
        let res = responseType === 'TEXT' ? xhr.responseText : xhr.response
        if (responseType === 'TEXT' && dataType === 'JSON') {
            try {
                res = JSON.parse(res)
            } catch (error) {
                // 和微信一致解析失败不抛出错误
                // invoke(callbackId, {
                //   errMsg: 'request:fail json parse error'
                // })
                // return
            }
        }
        invoke(callbackId, {
            data: res,
            statusCode,
            header: parseHeaders(xhr.getAllResponseHeaders()),
            errMsg: 'request:ok'
        })
    }
    xhr.onabort = function () {
        clearTimeout(timer)
        invoke(callbackId, {
            errMsg: 'request:fail abort'
        })
    }
    xhr.onerror = function () {
        clearTimeout(timer)
        invoke(callbackId, {
            errMsg: 'request:fail'
        })
    }
    xhr.send(body)
    return requestTask
}

```
