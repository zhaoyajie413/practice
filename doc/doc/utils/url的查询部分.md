  url查询部分的获取
  ====================
  
  ```
  function queryMap(url) {
          if (typeof url === 'string') {
              const urls = url.split('?')
  
              url = urls[0]
  
              const params = {};
              (urls[1] || '').split('&').forEach(function (pair) {
                  if (pair) {
                      const pairs = pair.split('=')
                      params[pairs[0]] = encodeURIComponent(pairs[1])
                  }
              })
  
              return params
          }
          return url
      }
      
  ```
