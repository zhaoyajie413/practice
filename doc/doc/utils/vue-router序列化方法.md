vue-router查询序列化
=========================
>code:
```angular2

    function stringifyQuery (obj) {
        var res = obj ? Object.keys(obj).map(function (key) {
            var val = obj[key];

            if (val === undefined) {
                return ''
            }

            if (val === null) {
                return encode(key)
            }

            if (Array.isArray(val)) {
                var result = [];
                val.forEach(function (val2) {
                    if (val2 === undefined) {
                        return
                    }
                    if (val2 === null) {
                        result.push(encode(key));
                    } else {
                        result.push(encode(key) + '=' + encode(val2));
                    }
                });
                return result.join('&')
            }

            return encode(key) + '=' + encode(val)
        }).filter(function (x) { return x.length > 0; }).join('&') : null;
        return res ? ("?" + res) : ''
    }

```
