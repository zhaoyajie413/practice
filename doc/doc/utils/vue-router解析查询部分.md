vue-router解析查询部分
======================

code:

```angular2
  function parseQuery (query) {
        var res = {};

        query = query.trim().replace(/^(\?|#|&)/, '');

        if (!query) {
            return res
        }

        query.split('&').forEach(function (param) {
            var parts = param.replace(/\+/g, ' ').split('=');
            console.log(parts)
            var key = decodeURIComponent(parts.shift());
            console.log(key)
            var val = parts.length > 0
                ?  decodeURIComponent(parts.join('='))
                : null;
            if (res[key] === undefined) {
                res[key] = val;
            } else if (Array.isArray(res[key])) {
                res[key].push(val);
            } else {
                res[key] = [res[key], val];
            }
        });

        return res
    }
```
