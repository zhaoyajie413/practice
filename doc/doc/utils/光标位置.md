获取光标的位置
===============

>获取div的光标

   ```angular2
//获取当前光标位置
const getCursortPosition = function (element) {
  var caretOffset = 0;
  var doc = element.ownerDocument || element.document;
  var win = doc.defaultView || doc.parentWindow;
  var sel;
  if (typeof win.getSelection != "undefined") {//谷歌、火狐
    sel = win.getSelection();
    if (sel.rangeCount > 0) {//选中的区域
      var range = win.getSelection().getRangeAt(0);
      var preCaretRange = range.cloneRange();//克隆一个选中区域
      preCaretRange.selectNodeContents(element);//设置选中区域的节点内容为当前节点
      preCaretRange.setEnd(range.endContainer, range.endOffset);  //重置选中区域的结束位置
      caretOffset = preCaretRange.toString().length;
    }
  } else if ((sel = doc.selection) && sel.type != "Control") {//IE
    var textRange = sel.createRange();
    var preCaretTextRange = doc.body.createTextRange();
    preCaretTextRange.moveToElementText(element);
    preCaretTextRange.setEndPoint("EndToEnd", textRange);
    caretOffset = preCaretTextRange.text.length;
  }
  return caretOffset;
}
```

> input/textarea获取光标位置

```angular2
//输入框获取光标
const getPosition = function (element) {
    let cursorPos = 0;
    if (document.selection) {//IE
        var selectRange = document.selection.createRange();
        selectRange.moveStart('character', -element.value.length);
        cursorPos = selectRange.text.length;
    } else if (element.selectionStart || element.selectionStart == '0') {
        cursorPos = element.selectionStart;
    }
    return cursorPos;
}

```

>可编辑div设置光标位置

```angular2
//设置光标位置
const setCaretPosition = function (element, pos) {
  var range, selection;
  if (document.createRange)//Firefox, Chrome, Opera, Safari, IE 9+
  {
    range = document.createRange();//创建一个选中区域
    range.selectNodeContents(element);//选中节点的内容
    if(element.innerHTML.length > 0) {
      range.setStart(element.childNodes[0], pos); //设置光标起始为指定位置
    }
    range.collapse(true);       //设置选中区域为一个点
    selection = window.getSelection();//获取当前选中区域
    selection.removeAllRanges();//移出所有的选中范围
    selection.addRange(range);//添加新建的范围
  }
  else if (document.selection)//IE 8 and lower
  {
    range = document.body.createTextRange();//Create a range (a range is a like the selection but invisible)
    range.moveToElementText(element);//Select the entire contents of the element with the range
    range.collapse(false);//collapse the range to the end point. false means collapse to end rather than the start
    range.select();//Select the range (make it the visible selection
  }
}
```

>input/textarea获取光标位置

```angular2
// 设置光标位置
function setCaretPosition(textDom, pos){
    if(textDom.setSelectionRange) {
        // IE Support
        textDom.focus();
        textDom.setSelectionRange(pos, pos);
    }else if (textDom.createTextRange) {
        // Firefox support
        var range = textDom.createTextRange();
        range.collapse(true);
        range.moveEnd('character', pos);
        range.moveStart('character', pos);
        range.select();
    }
}
```
