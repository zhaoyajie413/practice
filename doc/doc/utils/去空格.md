字符串去除空格
===================

```angular2   
function trim(value) { 
        return isString(value) ? value.replace(/^\s*/, '').replace(/\s*$/, '') : value;
 }
```
