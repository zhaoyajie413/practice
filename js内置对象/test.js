var strint = require("./strint");
var num1,num2;
		num1='89860012019100000060'
		num2='89860012019100000068'
		
		console.log(strint.sub(num1, num2))