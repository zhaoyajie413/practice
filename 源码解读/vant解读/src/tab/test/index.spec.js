import Vue from 'vue';
import Tab from '..';
import Tabs from '../../tabs';
import { mount, later, triggerDrag } from '../../../test/utils';

Vue.use(Tab);
Vue.use(Tabs);

function createWrapper(options = {}) {
  return mount({
    template: `
      <van-tabs
        :color="color"
        :type="type"
        :sticky="sticky"
        :line-width="lineWidth"
        :lazy-render="lazyRender"
        @change="onChange"
      >
        ${options.extraTemplate || ''}
        <van-tab title="title1">Text</van-tab>
        <van-tab title="title2">Text</van-tab>
        <van-tab title="title3" disabled>Text</van-tab>
      </van-tabs>
    `,
    data() {
      return {
        color: '#f44',
        type: 'line',
        sticky: true,
        lineWidth: 2,
        lazyRender: true
      };
    },
    ...options
  });
}

test('click to switch tab', async () => {
  const onChange = jest.fn();
  const wrapper = mount({
    template: `
      <van-tabs @change="onChange">
        <van-tab title="title1">Text</van-tab>
        <van-tab title="title2">Text</van-tab>
        <van-tab title="title3" disabled>Text</van-tab>
      </van-tabs>
    `,
    methods: {
      onChange
    }
  });

  await later();
  expect(wrapper).toMatchSnapshot();

  const tabs = wrapper.findAll('.van-tab');
  tabs.at(1).trigger('click');
  tabs.at(2).trigger('click');
  await later();
  expect(wrapper).toMatchSnapshot();
  expect(onChange).toHaveBeenCalledTimes(1);
});

test('swipe to switch tab', async () => {
  const onChange = jest.fn();
  const wrapper = mount({
    template: `
      <van-tabs swipeable @change="onChange">
        <van-tab title="title1">Text</van-tab>
        <van-tab title="title2">Text</van-tab>
        <van-tab title="title3" disabled>Text</van-tab>
      </van-tabs>
    `,
    methods: {
      onChange
    }
  });

  const content = wrapper.find('.van-tabs__content');
  await later();
  expect(wrapper).toMatchSnapshot();

  triggerDrag(content, -100, 0);
  expect(wrapper).toMatchSnapshot();
  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith(1, 'title2');

  triggerDrag(content, -100, 0);
  expect(onChange).toHaveBeenCalledTimes(1);
  expect(wrapper).toMatchSnapshot();

  await later();
  wrapper.destroy();
});

test('change tabs data', async () => {
  const wrapper = createWrapper();

  await later();

  expect(wrapper).toMatchSnapshot();

  wrapper.setData({
    swipeable: false,
    sticky: false,
    type: 'card',
    color: 'blue'
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});

test('lazy render', async () => {
  const wrapper = createWrapper();

  expect(wrapper).toMatchSnapshot();

  wrapper.setData({
    lazyRender: false
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});

test('render nav-left & nav-right slot', async () => {
  const wrapper = createWrapper({
    extraTemplate: `
      <template v-slot:nav-left>Nav Left</template>
      <template v-slot:nav-right>Nav Right</template>
    `
  });

  expect(wrapper).toMatchSnapshot();
});

test('border props', async () => {
  const wrapper = mount(Tabs, {
    propsData: {
      border: false
    }
  });

  expect(wrapper).toMatchSnapshot();
});


test('click event', async () => {
  const onClick = jest.fn();
  const onDisabled = jest.fn();

  const wrapper = mount({
    template: `
      <van-tabs @click="onClick" @disabled="onDisabled">
        <van-tab title="title1">Text</van-tab>
        <van-tab title="title2" disabled>Text</van-tab>
      </van-tabs>
    `,
    methods: {
      onClick,
      onDisabled
    }
  });

  const tabs = wrapper.findAll('.van-tab');

  tabs.at(0).trigger('click');
  expect(onClick).toHaveBeenCalledWith(0, 'title1');

  tabs.at(1).trigger('click');
  expect(onDisabled).toHaveBeenCalledWith(1, 'title2');
});


test('name prop', async () => {
  const onClick = jest.fn();
  const onChange = jest.fn();
  const onDisabled = jest.fn();

  const wrapper = mount({
    template: `
      <van-tabs @click="onClick" @disabled="onDisabled" @change="onChange">
        <van-tab title="title1" name="a">Text</van-tab>
        <van-tab title="title2" name="b">Text</van-tab>
        <van-tab title="title3" name="c" disabled>Text</van-tab>
      </van-tabs>
    `,
    methods: {
      onClick,
      onChange,
      onDisabled
    }
  });

  await later();
  expect(wrapper).toMatchSnapshot();

  const tabs = wrapper.findAll('.van-tab');
  tabs.at(1).trigger('click');

  expect(onClick).toHaveBeenCalledWith('b', 'title2');
  expect(onChange).toHaveBeenCalledWith('b', 'title2');
  expect(onChange).toHaveBeenCalledTimes(1);

  tabs.at(2).trigger('click');
  expect(onDisabled).toHaveBeenCalledWith('c', 'title3');
  expect(onChange).toHaveBeenCalledTimes(1);
});
