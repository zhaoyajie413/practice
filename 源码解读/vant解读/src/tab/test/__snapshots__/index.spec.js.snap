// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`border props 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div class="van-tabs__line"></div>
    </div>
  </div>
  <div class="van-tabs__content"></div>
</div>
`;

exports[`change tabs data 1`] = `
<div class="van-tabs van-tabs--line">
  <div>
    <div class="van-sticky">
      <div class="van-tabs__wrap van-hairline--top-bottom">
        <div role="tablist" class="van-tabs__nav van-tabs__nav--line" style="border-color: #f44;">
          <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
          <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
          <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
          <div class="van-tabs__line" style="background-color: rgb(255, 68, 68);"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`change tabs data 2`] = `
<div class="van-tabs van-tabs--card">
  <div class="van-tabs__wrap">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--card" style="border-color: blue;">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active" style="border-color: blue; background-color: blue;"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab" style="border-color: blue; color: blue;"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled" style="border-color: blue;"><span class="van-ellipsis">title3</span></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`click to switch tab 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`click to switch tab 2`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" class="van-tab"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab van-tab--active" aria-selected="true"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%); transition-duration: 0.3s;"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="display: none;">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`lazy render 1`] = `
<div class="van-tabs van-tabs--line">
  <div>
    <div class="van-sticky">
      <div class="van-tabs__wrap van-hairline--top-bottom">
        <div role="tablist" class="van-tabs__nav van-tabs__nav--line" style="border-color: #f44;">
          <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
          <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
          <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
          <div class="van-tabs__line" style="background-color: rgb(255, 68, 68);"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`lazy render 2`] = `
<div class="van-tabs van-tabs--line">
  <div>
    <div class="van-sticky">
      <div class="van-tabs__wrap van-hairline--top-bottom">
        <div role="tablist" class="van-tabs__nav van-tabs__nav--line" style="border-color: #f44;">
          <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
          <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
          <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
          <div class="van-tabs__line" style="background-color: rgb(255, 68, 68); width: 2px; transform: translateX(0px) translateX(-50%);"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">Text</div>
  </div>
</div>
`;

exports[`name prop 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`render nav-left & nav-right slot 1`] = `
<div class="van-tabs van-tabs--line">
  <div>
    <div class="van-sticky">
      <div class="van-tabs__wrap van-hairline--top-bottom">
        <div role="tablist" class="van-tabs__nav van-tabs__nav--line" style="border-color: #f44;">Nav Left<div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
          <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
          <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
          <div class="van-tabs__line" style="background-color: rgb(255, 68, 68);"></div>Nav Right
        </div>
      </div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 1`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 2`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="display: none;">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 3`] = `
<div class="van-tabs van-tabs--line">
  <div class="van-tabs__wrap van-hairline--top-bottom">
    <div role="tablist" class="van-tabs__nav van-tabs__nav--line">
      <div role="tab" aria-selected="true" class="van-tab van-tab--active"><span class="van-ellipsis">title1</span></div>
      <div role="tab" class="van-tab"><span class="van-ellipsis">title2</span></div>
      <div role="tab" class="van-tab van-tab--disabled"><span class="van-ellipsis">title3</span></div>
      <div class="van-tabs__line"></div>
    </div>
  </div>
  <div class="van-tabs__content">
    <div role="tabpanel" class="van-tab__pane" style="display: none;">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="">Text</div>
    <div role="tabpanel" class="van-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;
